ddtrace/tracer/sqlcomment.go

```diff
@@ -0,0 +1,174 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package tracer
+
+import (
+	"fmt"
+	"net/url"
+	"sort"
+	"strconv"
+	"strings"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+)
+
+// SQLCommentInjectionMode represents the mode of sql comment injection.
+type SQLCommentInjectionMode int
+
+const (
+	// CommentInjectionDisabled represents the comment injection mode where all injection is disabled.
+	CommentInjectionDisabled SQLCommentInjectionMode = 0
+	// ServiceTagsInjection represents the comment injection mode where only service tags (name, env, version) are injected.
+	ServiceTagsInjection SQLCommentInjectionMode = 1
+	// FullSQLCommentInjection represents the comment injection mode where both service tags and tracing tags. Tracing tags include span id, trace id and sampling priority.
+	FullSQLCommentInjection SQLCommentInjectionMode = 2
+)
+
+// Key names for SQL comment tags.
+const (
+	SamplingPrioritySQLCommentKey   = "ddsp"
+	TraceIDSQLCommentKey            = "ddtid"
+	SpanIDSQLCommentKey             = "ddsid"
+	ServiceNameSQLCommentKey        = "ddsn"
+	ServiceVersionSQLCommentKey     = "ddsv"
+	ServiceEnvironmentSQLCommentKey = "dde"
+)
+
+// SQLCommentCarrier is a carrier implementation that injects a span context in a SQL query in the form
+// of a sqlcommenter formatted comment prepended to the original query text.
+// See https://google.github.io/sqlcommenter/spec/ for more details.
+type SQLCommentCarrier struct {
+	Query  string
+	Mode   SQLCommentInjectionMode
+	SpanID uint64
+}
+
+// NewSQLCommentCarrier returns a new instance of a SQLCommentCarrier
+func NewSQLCommentCarrier(query string, mode SQLCommentInjectionMode) (c *SQLCommentCarrier) {
+	c = new(SQLCommentCarrier)
+	c.Mode = mode
+	c.Query = query
+	c.SpanID = random.Uint64()
+	return c
+}
+
+// Inject injects a span context in the carrier's query.
+func (c *SQLCommentCarrier) Inject(spanCtx ddtrace.SpanContext) error {
+	if c.Mode == CommentInjectionDisabled {
+		return nil
+	}
+
```

Comment:
nit: we don't need this space. Also, the above `if` clause can go into the `switch` too.